harness/determined/_data_layer/_data_layer.py

```diff
@@ -59,11 +59,7 @@ def _init_offset(self) -> None:
             return
 
         batch_size = self._per_slot_batch_size
-        batches_completed = self._env.first_step() - 1
-        if self._env.initial_workload.kind == workload.Workload.Kind.COMPUTE_VALIDATION_METRICS:
-            batches_completed += 1
-
-        self._offset = batches_completed * batch_size
+        self._offset = self._env.initial_workload.total_batches_processed * batch_size
```

Comment:
@rb-determined-ai (maybe @aaron276h) does this look right. just noticed it while I was doing the last rename and, didn't dig too deep but feels ok (making a ton of assumptions, also the original code looks like it's calculating step_id * per_slot_batch_size, which feels weird (aka doesn't make sense to me) anyways).

Reply:
Yeah this was definitely wrong before, nice catch!